### DISCUSSION TITLE
Control legs separately

### DISCUSSION THREAD

**Discussion by kissgyula** [gcormier/megadesk#142] [General]
Is it possible to control the legs separately with this modification? I don't specifically want to use it as a table leg, and I would need it to be able to adjust the two legs completely independently. In addition, I would also like to know if it is possible to use it with only one leg connected?

**Comment by gcormier** [comment]
Out of the box I doubt it will work. But with some code modifications, it should absolutely work with 1 leg only.

**Reply by kissgyula** [reply]
Thank you for the answer!

I try to study the code, and find there is node_a and node_b, and also enc_a and enc_b where the lin.recv reads the motor states as I understand.

But the lin.send part is what I not understand fully. Is the cmd[0] = enc_target & 0xFF; and cmd[1] = enc_target >> 8; around line 1062 in the code, will set the target for the two motors? If I set this to two different encoder targets, the legs will move to a different position?

**Reply by AnAnalogGuy** [reply]
Controlling one leg alone potentially is possible by just connecting one leg of the table to the control unit. 
But i don't see how two connected legs could be controlled independently by one control unit, as far as I understand the electrical circuit of the table, the legs are not controlled independently.

**Comment by gcormier** [comment]
https://github.com/gcormier/megadesk/blob/e1840cca942da1b69034d85871321d3d74baeedd/code/src/megadesk.cpp#L992-L997

This is where megadesk sends the command (PID 18).

Having 1 motor would not change anything as the motors are listening for this command.

The only time the individual legs come into play is when they respond to the height query for their unique PID. https://github.com/gcormier/megadesk/blob/e1840cca942da1b69034d85871321d3d74baeedd/code/src/megadesk.cpp#L818-L843

This is where we check the height of both motors, and subsequently use min/max values between the two to decide which is going to determine our actual height. You would want to only query, say, PID 8 and use that for the current height status.
